hpx/runtime/threads/detail/scheduling_loop.hpp

```diff
@@ -484,10 +503,19 @@ namespace hpx { namespace threads { namespace detail
                 // do background work in parcel layer and in agas
                 if ((scheduler.get_scheduler_mode() & policies::do_background_work) &&
                     num_thread < callbacks.max_background_threads_ &&
-                    !callbacks.background_.empty())
+                    background_thread)
                 {
-                    if (callbacks.background_())
-                        idle_loop_count = 0;
+                    thread_result_type background_result = (*background_thread)();
+                    if (next_thrd == nullptr)
+                    {
+                        next_thrd = background_result.second.get();
+                    }
+                    else if(background_result.second != nullptr &&
+                        background_result.second != background_thread)
+                    {
+                        scheduler.SchedulingPolicy::schedule_thread(
+                            background_result.second.get(), num_thread);
+                    }
                 }
```

Comment:
It currently is nullptr, yes, but what if some background_result uses `suspend_boost` or yields a thread to be scheduled?